dot/parachain/backing/validated_candidate_command.go

```diff
@@ -268,12 +268,12 @@ func (cb *CandidateBacking) handleCommandAttest(
 		return nil
 	}
 
-	statement := parachaintypes.NewStatementVDT()
-	if err := statement.SetValue(parachaintypes.Valid(candidateHash)); err != nil {
-		return fmt.Errorf("setting statement: %w", err)
-	}
-
 	if bgValidationResult.err == nil {
+		statement := parachaintypes.NewStatementVDT()
+		if err := statement.SetValue(parachaintypes.Valid(candidateHash)); err != nil {
+			return fmt.Errorf("setting statement: %w", err)
+		}
+
```

Comment:
We need a statement only when there is no error in validation.
So, I have just moved the code.